state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -409,8 +414,7 @@ pub mod pallet {
 			));
 
 			let (debit_amount, redeem_amount) = match amount {
-				RedemptionAmount::Max =>
-					(liquid_balance, liquid_balance.saturating_sub(redemption_fee)),
+				RedemptionAmount::Max => (liquid_balance, liquid_balance),
```

Comment:
I think what we need to do is something like this: 

```suggestion
				RedemptionAmount::Max => {
					if liquid_balance == T::Flip::balance(&account_id) {
						// Withdrawing all -> no tax
						liquid_balance
					} else {
						liquid_balance.saturating_sub(redemption_fee)
					}
				},
```